python/idsse_common/idsse/common/rabbitmq_utils.py

```diff
@@ -358,36 +460,44 @@ class Consumer(Thread):
     shutdown.  The start() and stop() methods should be called from the same
     thread as the one used to create the instance.
     """
-
-    # pylint: disable=too-many-instance-attributes
-    # Eight is reasonable in this case.
-
     def __init__(
         self,
         conn_params: Conn,
         rmq_params_and_callbacks: RabbitMqParamsAndCallback | list[RabbitMqParamsAndCallback],
-        num_message_handlers: int,
         *args,
+        num_message_handlers: int = 2,
         **kwargs,
     ):
-        super().__init__(*args, **kwargs)
+        """
+        Args:
+            conn_params (Conn): parameters to create a new RabbitMQ connection
+            rmq_params_and_callbacks (RabbitMqParamsAndCallback | list[RabbitMqParamsAndCallback]):
+                1 or more Exch/Queue tuples, and a function to invoke when messages arrive on the
+                listed queue.
+            num_message_handlers (optional, int): The max thread pool size for workers to handle
+                message callbacks concurrently. Default is 2.
+        """
+        super().__init__(*args, **kwargs, name='Consumer')
         self.context = contextvars.copy_context()
         self.daemon = True
         self._tpx = ThreadPoolExecutor(max_workers=num_message_handlers)
-        self._conn_params = conn_params
+
         if isinstance(rmq_params_and_callbacks, list):
-            self._rmq_params_and_callbacks = rmq_params_and_callbacks
```

Comment:
Did you drop the 'self.' because there is no need to store after initializing the instants, or is there another reason for the change?

Reply:
Yep, I just realized this wasn't used anywhere else in the class besides `__init__`.